azure-container-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/containerapps/parser/ConfigParser.java

```diff
@@ -30,22 +39,31 @@ public ContainerAppConfig getContainerAppConfig() {
         config.setAppName(mojo.getAppName());
         config.setResourceConfiguration(getResourceConfigurationFromContainers(mojo.getContainers()));
         config.setIngressConfig(getIngressConfig(mojo.getIngress()));
-        config.setImageConfig(getImageConfigFromContainers(mojo.getContainers()));
+        config.setRegistryConfig(getRegistryConfig());
+        config.setImageConfig(getImageConfigFromContainers());
         config.setScaleConfig(mojo.getScale());
         return config;
     }
 
 
     //todo(ruoyuwang): whether we should support multi containers? CLI/Portal only support one container
     //todo(ruoyuwang): support container registry part
-    public ContainerAppDraft.ImageConfig getImageConfigFromContainers(List<AppContainerMavenConfig> containers) {
+    public ContainerAppDraft.ImageConfig getImageConfigFromContainers() {
+        List<AppContainerMavenConfig> containers = mojo.getContainers();
         if (containers == null || containers.isEmpty()) {
             return null;
         }
-        final ContainerAppDraft.ImageConfig imageConfig = new ContainerAppDraft.ImageConfig(containers.get(0).getImage());
+        final String defaultImageName = String.format("%s%s/%s:%s", mojo.getRegistry().getRegistryName(), ContainerRegistry.ACR_IMAGE_SUFFIX, mojo.getAppName(), timestamp);
+        final String fullImageName = Optional.ofNullable(containers.get(0).getImage()).orElse(defaultImageName);
```

Comment:
In CI scenario, it's not a good practice to reuse the same name. The container environment may set `imagePullPolicy=IfNotPresent`, which will prevent further update of the image with the same tag.

Better to check with tooling team on recommended experience here.

Reply:
Customers are likely to provide no image part in their config when using code/artifact type. We will generate the `defaultImageName` for them and won't reuse the name. The code just gives them an option to specify it. Most of the time, this scenario doesn't happen.